ESP_socket_embedded_motors/src/main.cpp

```diff
@@ -0,0 +1,616 @@
+//------------------------------------------------------------------------------------------------//
+//
+//
+//-------------------------------------------------//
+//                                                 //
+//    /////////      //            //////////      //
+//    //      //     //            //        //    //
+//    //      //     //            //        //    //
+//    ////////       //            //////////      //
+//    //      //     //            //              //
+//    //       //    //            //              //
+//    /////////      //////////    //              //
+//                                                 //
+//-------------------------------------------------//
+void loop() {
+    //----------//
+    //  Timers  //
+    //----------//
+
+    // Blink the LED
+    if (millis() - lastBlink >= 1000) 
+    {
+        lastBlink = millis();
+        ledState = !ledState;
+        if (ledState)
+            digitalWrite(LED_BUILTIN, HIGH);
+        else
+            digitalWrite(LED_BUILTIN, LOW);
+    }
+
+    // Accelerate motors; update the speed for all motors
+    if (millis() - lastAccel >= 100) 
+    {
+        lastAccel = millis();
+        for (int i = 0; i < MOTOR_AMOUNT; i++) 
+        {
+            motorList[i]->accelerate();
+        }
+    }
+
+    // Heartbeat for REV motors
+    if (millis() - lastHB >= 3)
+    {
+        sendHeartbeat(ESP32Can, heartBeatNum);
+        lastHB = millis();
+        heartBeatNum++;
+        if (heartBeatNum > 4)
+        {
+            heartBeatNum = 1;
+        }
+    }
+
+    if (millis() - lastMotorFeedback >= 2000) // Change to 1000?
+    {
+        lastMotorFeedback = millis();
+        motorFeedback();
+    }
+
+    // Safety timeout
+    safety_timeout();
+
+    // Motor status debug printout
+#if defined(DEBUG2)
+    if (millis() - lastMotorStatus > 500) {
+        lastMotorStatus = millis();
+
+        // Status 1
+        Serial.print(millis() - Motor2.status1.timestamp);
+        Serial.print(" ms ago: ");
+        Serial.print(Motor2.status1.motorTemperature);
+        Serial.print(" *C; ");
+        Serial.print(Motor2.status1.busVoltage);
+        Serial.print(" V; ");
+        Serial.print(Motor2.status1.outputCurrent);
+        Serial.print(" A; ");
+        Serial.print(Motor2.status1.sensorVelocity);
+        Serial.println(" RPM");
+        
+        // Status 2
+        Serial.print(millis() - Motor2.status2.timestamp);
+        Serial.print(" ms ago: ");
+        Serial.print(Motor2.status2.sensorPosition);
+        Serial.println(" rotations");
+    }
+#endif
+
+    //-------------//
+    //  CAN Input  //
+    //-------------//
+
+    static CanFrame rxFrame;
+    if (ESP32Can.readFrame(rxFrame, 1)) {
+        // Decode the ID
+
+        uint32_t msgId = rxFrame.identifier;
+        // Pull out device ID and API ID
+        uint8_t deviceId = msgId & 0x3F;
+        uint32_t apiId = (msgId >> 6) & 0x3FF;
+
+        if (apiId == 0x60) {
```

Comment:
all of these apiId magic numbers should be commented. perhaps a reference to rover-Docs?